### PR TITLE
Fixes #10685: Add Technique editor documentation

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://www.rudder-project.org/redmine/issues/10685

### CODE REVIEW

File: 22_configuration_management/38_technique_editor.txt [hunk 1]
```diff
@@ -0,0 +1,85 @@
+=== Technique editor
```
Comment (fanf) [inline 22_configuration_management/38_technique_editor.txt]: Ajouter avant le titre l'ancre: [[technique-editor]]

File: 22_configuration_management/38_technique_editor.txt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+==== Introduction
+
+===== First, what is a Technique ?
+A technique is a description in code form of what the agent has to do on the node. This code is actually composed of a series of Generic method calls. These different Generic method calls are conditional.
```
Comment (fanf) [inline 22_configuration_management/38_technique_editor.txt]: line break after each title

File: 22_configuration_management/38_technique_editor.txt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+==== Technique Editor
+
+===== Utility
+Rudder provides a set of pre-defined Techniques that cover some basic configuration and system administration needs. Of course,this set of techniques cannot responds to all of the specific needs of each client. That’s why Rudder integrate the *Technique editor*, a tool to create advanced Techniques.
+Directly accessible from Ruder menu (_Utilities > Technique editor_), this tool has an easy-to-use interface, which doesn’t require any programming skills but nevertheless allows to create complex Techniques.
+
+===== Interface
+Here is an overview of its interface :
+
+image::./images/technique_editor/1-rudder-technique-editor.png[]
```
Comment (fanf) [inline 22_configuration_management/38_technique_editor.txt]: line break after image

File: 22_configuration_management/38_technique_editor.txt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+The interface is divided into 3 columns:
+
+
+.1. A column listing custom Techniques
+image::./images/technique_editor/2-list-techniques.png[]
+Here, we can see our previously created Techniques. We can click on them to see their details/edit them, or create a new one by clicking on the “New” button. Theses Techniques are visible in the *ncf techniques* category in the *Active Techniques tree*, so can be used to create new Directives.
```
Comment (VinceMacBuche) [inline 22_configuration_management/38_technique_editor.txt]: Maybe we should not talk about the "active techniques tree" but only in the "Directive tree" ?

File: 22_configuration_management/38_technique_editor.txt [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+image::./images/technique_editor/5-configure-generic-method.png[]
+
+The Generic method details are divided into 3 blocks :
+.Conditions
+Conditions allow user to restrict the execution of the method. 
+.Parameters
+Parameters are in mono or multi line text format. They can contains variables which will be extended at the time of the execution.(syntax:${xxxx})
+.Result classes
+One result class of three will be defined following the execution of a generic method:
+ * One, in case of no modification (Success)
```
Comment (VinceMacBuche) [inline 22_configuration_management/38_technique_editor.txt]: You could say "Success, when the configuration is correct and no action are needed"

File: 22_configuration_management/38_technique_editor.txt [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+ * One, in case of modification (Repaired)
```
Comment (VinceMacBuche) [inline 22_configuration_management/38_technique_editor.txt]: "Repaired, when the configuration is wrong and actions to fix it were executed with success"

File: 22_configuration_management/38_technique_editor.txt [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+ * One, in case of error (Error)
```
Comment (VinceMacBuche) [inline 22_configuration_management/38_technique_editor.txt]: Error, when the configuration is wrong but actions to fix it failed

File: 22_configuration_management/38_technique_editor.txt [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+Theses classes can be used in another Generic methods conditions.
```
Comment (VinceMacBuche) [inline 22_configuration_management/38_technique_editor.txt]: Maybe more details here ? "ie, you can execute a command if a previous one failed or was repaired"

File: 22_configuration_management/38_technique_editor.txt [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+Now, we have to find and add the generic methods which correspond to the actions we want to execute. In our case, we want to add the following methods:
+* Package install (You can find it in the *Package category*)
+  - This method only take one parameter, the name of the package to install. So here, fill in the *package_name* field with the value _ntp_.
+
+* File from template (You can find it in the *File category*)
+  - This method take two parameters. The first one corresponds to the absolute path of the source file containing a template to be expanded. We are going to use a Rudder variable here to get the correct path. Fill in the *source_template* field with the value _${path_technique}/templates/ntp.conf_.
+	The second corresponds to the absolute path of the destination file. Fill in with the value _/etc/ntp.conf_.
+
+* Service restart (You can find it in the *Package category*)
+  - This method only take one parameter, the name of the service we want to restart. So here, fill in the *service_name* field with the value _ntp_.
```
Comment (VinceMacBuche) [inline 22_configuration_management/38_technique_editor.txt]: You should put a condition here, either package was installed or configuration changed)

File: 22_configuration_management/38_technique_editor.txt [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+===== 2. Add and configure generic methods
+Now, we have to find and add the generic methods which correspond to the actions we want to execute. In our case, we want to add the following methods:
+* Package install (You can find it in the *Package category*)
+  - This method only take one parameter, the name of the package to install. So here, fill in the *package_name* field with the value _ntp_.
+
+* File from template (You can find it in the *File category*)
+  - This method take two parameters. The first one corresponds to the absolute path of the source file containing a template to be expanded. We are going to use a Rudder variable here to get the correct path. Fill in the *source_template* field with the value _${path_technique}/templates/ntp.conf_.
+	The second corresponds to the absolute path of the destination file. Fill in with the value _/etc/ntp.conf_.
+
+* Service restart (You can find it in the *Package category*)
```
Comment (VinceMacBuche) [inline 22_configuration_management/38_technique_editor.txt]: Service category

File: 22_configuration_management/38_technique_editor.txt [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+  - This method only take one parameter, the name of the service we want to restart. So here, fill in the *service_name* field with the value _ntp_.
+* Service ensure running (You can find it in the *Package category*)
```
Comment (VinceMacBuche) [inline 22_configuration_management/38_technique_editor.txt]: Service category

File: 22_configuration_management/38_technique_editor.txt [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+  - This method only take one parameter, the name of the service we want to check. Again, here, fill in the *service_name* field with the value _ntp_.
+
+===== 3. Done!
+And… It’s already done. Rather fast, right? Don’t forget to save you technique. Now you can see it in the *Active Techniques tree*, and  use that it to create a _Directive_ that will be applied on your _Nodes_ thanks to a _Rule_.
```
Comment (VinceMacBuche) [inline 22_configuration_management/38_technique_editor.txt]: you*r* technique, or just "Don't forget to save"

Comment (VinceMacBuche) [inline 22_configuration_management/38_technique_editor.txt]: Would also speak of Directive tree, so people go to Directive page and not technique one

File: 22_configuration_management/38_technique_editor.txt [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+* File from template (You can find it in the *File category*)
+  - This method take two parameters. The first one corresponds to the absolute path of the source file containing a template to be expanded. We are going to use a Rudder variable here to get the correct path. Fill in the *source_template* field with the value _${path_technique}/templates/ntp.conf_.
+	The second corresponds to the absolute path of the destination file. Fill in with the value _/etc/ntp.conf_.
+
+* Service restart (You can find it in the *Package category*)
+  - This method only take one parameter, the name of the service we want to restart. So here, fill in the *service_name* field with the value _ntp_.
+* Service ensure running (You can find it in the *Package category*)
+  - This method only take one parameter, the name of the service we want to check. Again, here, fill in the *service_name* field with the value _ntp_.
+
+===== 3. Done!
```
Comment (VinceMacBuche) [inline 22_configuration_management/38_technique_editor.txt]: Maybe rename "Done!" to "Save and apply your technique"

**Comment by RaphaelGauthier** [comment]
Commit modified

**Pull request merged**